### PR TITLE
add dropout to VelocityField and OTFM/GENOT

### PR DESCRIPTION
**Pull request by MUCDK**

### CODE REVIEW

File: src/ott/neural/networks/velocity_field.py [hunk 1]
```diff
@@ -52,20 +52,23 @@ class VelocityField(nn.Module):
...
   act_fn: Callable[[jnp.ndarray], jnp.ndarray] = nn.silu
+  dropout_rate: float = 0.0
 
   @nn.compact
   def __call__(
       self,
       t: jnp.ndarray,
       x: jnp.ndarray,
       condition: Optional[jnp.ndarray] = None,
+      deterministic: bool = False
```
Comment (michalk8) [inline src/ott/neural/networks/velocity_field.py:63]: Think it's better to use the `train: bool = True` naming (+ flipping the logic), more explicit.

File: src/ott/neural/methods/flows/genot.py [hunk 2]
```diff
@@ -286,7 +286,11 @@ def transport(
 
     def vf(t: jnp.ndarray, x: jnp.ndarray, cond: jnp.ndarray) -> jnp.ndarray:
       params = self.vf_state.params
-      return self.vf_state.apply_fn({"params": params}, t, x, cond)
+      return self.vf_state.apply_fn({"params": params},
+                                    t,
+                                    x,
+                                    cond,
+                                    deterministic=False)
```
Comment (michalk8) [inline src/ott/neural/methods/flows/genot.py:293]: Shouldn't this be `True` when transporting (as it's inference)?

File: src/ott/neural/methods/flows/otfm.py [hunk 3]
```diff
@@ -175,7 +175,11 @@ def vf(
         t: jnp.ndarray, x: jnp.ndarray, cond: Optional[jnp.ndarray]
     ) -> jnp.ndarray:
       params = self.vf_state.params
-      return self.vf_state.apply_fn({"params": params}, t, x, cond)
+      return self.vf_state.apply_fn({"params": params},
+                                    t,
+                                    x,
+                                    cond,
+                                    deterministic=False)
```
Comment (michalk8) [inline src/ott/neural/methods/flows/otfm.py:182]: Same here, shouldn't this be `True` when transporting (as it's inference)?

**Review by michalk8 — CHANGES_REQUESTED** [review]
Currently, in the training of OTFM/GENOT, we should pass explicitly `train=True` (even with will have `train=True` by default).
Based on [this](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/layers.html#flax.linen.Dropout), will require passing `{'dropout': jax.random.PRNGKey(...)}` alongside the params.
Not sure why it's not complaining right now, but I assume because we don't have a test for the VF with `dropout > 0` - could you please add?

**Review by michalk8 — APPROVED** [review]
Thanks @MUCDK , LGTM!
